cmake-modules/AzureCodeCoverage.cmake

```diff
@@ -55,7 +55,8 @@ function(add_gcovr_html)
             -r ${BASEDIR} ${EXCLUDE_ARGS}
             --object-directory=${PROJECT_BINARY_DIR}
             -o ${args_TARGET_NAME}/index.html
-
+            --exclude-unreachable-branches
+            --exclude-throw-branches
```

Comment:
These we should leave and add tests that will throw from those paths.

Reply:
"throw branches" are not "branches that contain throw directives". "throw branches" are branches which the compiler generates whenever you call a method which may throw.

So it's not feasible to write code to cover those branches, unless we figure out a way to convince the C++ standard libraries to throw.

There's more information in the [GCovr FAQ](https://gcovr.com/en/stable/faq.html#why-does-c-code-have-so-many-uncovered-branches)

Reply:
> With the [gcovr --exclude-throw-branches](https://gcovr.com/en/stable/manpage.html#cmdoption-gcovr-exclude-throw-branches) option, exception-only branches will be ignored. These are typically arcs from a function call into an exception handler.

> For branch coverage, exclude branches that the compiler generates for exception handling. This often leads to more ‘sensible’ coverage reports. Config key(s): exclude-throw-branches.



Interesting!